utils/post.js

```diff
@@ -40,8 +40,13 @@ const filterAllTopics = (text, topics = []) => {
     return acc;
   }, []);
 
-  return [...new Set([...topicsFromTextWithoutHashtag, ...topics])];
-  // return topics;
+  let unique_topics = [
+    ...new Set([
+      ...topicsFromTextWithoutHashtag.map((topic) => topic.toLowerCase()),
+      ...topics.map((topic) => topic.toLowerCase())
+    ])
+  ];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic of converting topics to lowercase before saving them to the database is correct. However, it's important to consider that this change might affect other parts of the system where case sensitivity matters. For example, if there are any operations that rely on the original casing of the topics, they may not work as expected after this change. 

If you're sure that this won't cause any issues, then the code looks good. But if there's a chance that it could, you might want to add a configuration option to control whether or not topics should be converted to lowercase. This way, you can easily switch between the two behaviors based on your needs.

Here's an example of how you could implement this:

```diff
  let unique_topics = [
    ...new Set([
      ...topicsFromTextWithoutHashtag.map((topic) => topic.toLowerCase()),
      ...topics.map((topic) => topic.toLowerCase())
    ])
  ];
+ if (config.convertTopicsToLowercase) {
+   unique_topics = unique_topics.map(topic => topic.toLowerCase());
+ }
  return unique_topics;
```

In this example, `config.convertTopicsToLowercase` would be a boolean value that you set based on your application's needs.


<!-- This is an auto-generated comment by OSS CodeRabbit -->